### ISSUE TITLE
We need more official documentation about plugins for modular systems

### ISSUE THREAD

**Issue by aecarlosae** [cakephp/docs#5989]
**Issue Description**

Hello guys. I have been searching for plugin documentation to build a modular system but I do not found enogh official docs for distincts aspects that a new developer need to know. For example, when a plugin is more that a simple functionality, when this one, include session, login process, custom session and cookies data, integration with global vendors sources... I think tht in [Plugins](https://book.cakephp.org/3.0/en/plugins.html) info, with need to cover much more.

**Comment by markstory** [comment]
Isn't a login process made of controllers, components, models and templates? Are the plugins you're trying to create contained within your application repository or are they separate repositories?

**Comment by aecarlosae** [comment]
> Isn't a login process made of controllers, components, models and templates? Are the plugins you're trying to create contained within your application repository or are they separate repositories?

They form a single application that must be created in a total modular way. I have been coding using CakePHP's plugins but (I think) there is not enough info about plugins for modular application development.

**Comment by markstory** [comment]
Ok. I don't know enough about what is missing to write those docs myself, but I can help edit or round out examples from others.

**Comment by aecarlosae** [comment]
> Ok. I don't know enough about what is missing to write those docs myself, but I can help edit or round out examples from others.

Ok, imagine a module that needs to use auth component in some actions. How do you use application auth component in a plugin that has a models, views and controllers?

**Comment by markstory** [comment]
I thought we had docs for that in the components chapter? https://book.cakephp.org/3.0/en/controllers.html#configuring-components-to-load

We could add links to that section from the plugin chapter as well if you think it would help.

**Comment by aecarlosae** [comment]
> I thought we had docs for that in the components chapter? https://book.cakephp.org/3.0/en/controllers.html#configuring-components-to-load
> 
> We could add links to that section from the plugin chapter as well if you think it would help.

Yes. I think that at least a link as a references or a little 'how to' is important to let us know how much modular a project can be build with.

**Comment by aecarlosae** [comment]
I am building a company admin system... It must be build in a modular way (CAKEPHP IS PERFECT FOR ME). I have to build 5 programs as a plugin (a module) each one of them. 

Each program (plugin or module) must share the same database, the same authentication process... to work together but they can work independently (Obviously)

My initial problem was to discover if cakephp was the right framework to implement or build this system with the official docs' plugin section... I did not found the enough infomation in it... I assumed that I had to apply the same for an app in the normal development way in cakephp.

Because that, I think is necesary to create more references about the use of framework's component, helpers... for other new developers Or at least highlight that everything, for a normal cakephp app, applies in the same way when you build a plugin.
